examples/shapes.rs

```diff
@@ -35,36 +35,23 @@ fn main() {
 
     let mut msphere = {
         let geometry = three::Geometry::uv_sphere(2.0, 5, 5);
-        let material = three::Material::MeshBasic {
+        let material = three::material::Basic {
             color: 0xff0000,
             map: None,
-            wireframe: true,
+            pipeline: three::material::basic::Pipeline::Wireframe,
         };
         win.factory.mesh(geometry, material)
     };
     msphere.set_position([-3.0, 3.0, 0.0]);
     win.scene.add(&msphere);
 
-    let mut mline = {
-        let geometry = three::Geometry::with_vertices(vec![
-            [-2.0, -1.0, 0.0].into(),
-            [0.0, 1.0, 0.0].into(),
-            [2.0, -1.0, 0.0].into(),
-        ]);
-        let material = three::Material::LineBasic { color: 0x0000ff };
-        win.factory.mesh(geometry, material)
-    };
-    mline.set_position([3.0, 3.0, 0.0]);
-    win.scene.add(&mline);
-
     let mut angle = cgmath::Rad::zero();
     while win.update() && !three::KEY_ESCAPE.is_hit(&win.input) {
         if let Some(diff) = three::AXIS_LEFT_RIGHT.timed(&win.input) {
             angle += cgmath::Rad(1.5 * diff);
             let q = cgmath::Quaternion::from_angle_y(angle);
             mbox.set_orientation(q);
             mcyl.set_orientation(q);
-            mline.set_orientation(q);
```

Comment:
why is line removed?

Reply:
Because support for rendering lines is removed.

Reply:
yes, but why?

Reply:
I don't feel it belongs in the API. The lines are not anti-aliased, they make the pipelines more complex, they don't match up at the ends, and they're seldom even used. I think it would be a better option to provide a line geometry constructor/builder or an explicit `LineMesh` type instead, although I'm not going to include that in this PR.

Reply:
Relevant: https://blog.mapbox.com/drawing-antialiased-lines-with-opengl-8766f34192dc

Reply:
I could add it back but rename the material just `Line` instead of `LineBasic`like we have the `Sprite` material.

Reply:
I don't find the arguments convincing. Lines are not ideal, but they do allow for user cases previously inaccessible, and they still use the same PSO definition so not much needed to support from our side. Please bring `Line` back in some form.

Reply:
Okay, will do. :smile: